src/main/webapp/app/overview/header-course.component.ts

```
@@ -9,28 +9,39 @@
 
+    // Removed ngOnInit to consolidate initialization logic
+
+    ngOnChanges() {
+        this.courseColor = this.course.color || ARTEMIS_DEFAULT_COLOR;
+        this.contentColor = getContrastingTextColor(this.courseColor);
+        this.adjustCourseDescription();
+    }
+
     ngOnChanges() {
         this.adjustCourseDescription();
     }
```

Biome (1.9.4)

[error] Duplicate class member name "ngOnChanges"



(lint/suspicious/noDuplicateClassMembers)